spec/controllers/admin/users_controller_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to specify the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator.  If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails.  There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+#
+# Compared to earlier versions of this generator, there is very limited use of
+# stubs and message expectations in this spec.  Stubs are only used when there
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+
+RSpec.describe Admin::UsersController, type: :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # User. As you add validations to User, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    {
+      first_name: "Tony",
+      last_name: "Stark",
+      email: "tony@example.com",
+      password: "password",
+      active: true,
+      role: "stockholder",
+    }
+  }
+
+  let(:invalid_attributes) {
+    {
+      first_name: "Tony",
+      email: "tony@example",
+    }
+  }
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # UsersController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all users as @users" do
+      user = User.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:users)).to eq([user])
+    end
+  end
+
+  describe "GET #new" do
+    it "assigns a new user as @user" do
+      get :new, {}, valid_session
+      expect(assigns(:user)).to be_a_new(User)
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested user as @user" do
+      user = User.create! valid_attributes
+      get :edit, {:id => user.to_param}, valid_session
+      expect(assigns(:user)).to eq(user)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new User" do
+        expect {
+          post :create, {:user => valid_attributes}, valid_session
+        }.to change(User, :count).by(1)
+      end
+
+      it "assigns a newly created user as @user" do
+        post :create, {:user => valid_attributes}, valid_session
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.